### PR TITLE
UI test : Covered tests for Policy view flow of read only user

### PR DESCRIPTION
**Pull request by RusJaI**
This completes the UI test flow of #274 which was partially blocked with [this issue](https://github.com/wso2/api-manager/issues/151).Some minor changes added to avoid random failures of tests.

### CODE REVIEW

**Review by chanaka3d — APPROVED** [review]

File: tests/cypress/integration/publisher/019-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js [hunk 1]
```diff
@@ -233,7 +236,15 @@ describe("Publisher Read-Only Mode", () => {
         cy.get('table').get('tbody').get('[data-testid="MUIDataTableBodyRow-0"]').get('[data-testid="MuiDataTableBodyCell-4-0"]').get('[aria-label="Edit creatorscope"]').get('[aria-disabled="true"]').should('exist');
         cy.get('table').get('tbody').get('[data-testid="MUIDataTableBodyRow-0"]').get('[data-testid="MuiDataTableBodyCell-4-0"]').contains('button','Delete').should('be.disabled');
 
-        //12. Policies should be checked. for now cannot due to a UI bug
+        //12. Policies should be checked. (UI issue fixed by PR #11297 in carbon-apimgt)
+        cy.get("#left-menu-policies").click();
+        cy.get('[data-testid="add-new-api-specific-policy"]').click();
+        cy.get('[data-testid="create-policy-form"]').get('[data-testid="displayname"]').type("test name");
+        cy.get('[data-testid="create-policy-form"]').get('[data-testid="gateway-details-panel"]').get('[data-testid="file-drop-zone"]').then(function () {
```
Comment (sajinieKavindya) [inline tests/cypress/integration/publisher/019-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js:243]: For readability, avoid longer lines.

File: tests/cypress/integration/publisher/019-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js [hunk 2]
```diff
@@ -67,24 +67,27 @@ describe("Publisher Read-Only Mode", () => {
...
-        const dataTransfer = new DataTransfer();
-        cy.contains('Add Header').trigger('dragstart',{
-            dataTransfer
+        cy.location('pathname').then((pathName) => {
+            const pathSegments = pathName.split('/');
+            const uuid = pathSegments[pathSegments.length - 2];
+            cy.visit(`${Utils.getAppOrigin()}/publisher/apis/${uuid}/policies`);
+
+            const dataTransfer = new DataTransfer();
+            cy.contains('Add Header').trigger('dragstart',{
```
[the remark this answers is not in the document]
Comment (RusJaI) [reply tests/cypress/integration/publisher/019-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js:76]: solved

File: tests/cypress/integration/publisher/019-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js (hunk 1 above)
Comment (RusJaI) [reply tests/cypress/integration/publisher/019-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js:243]: solved

File: tests/cypress/integration/publisher/019-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js [hunk 3]
```diff
@@ -67,24 +67,27 @@ describe("Publisher Read-Only Mode", () => {
...
+                dataTransfer
+            });
+            cy.get('#headerName').type('Testing');
+            cy.get('#headerValue').type('abc');
+            cy.get('[data-testid="policy-attached-details-save"]').click();
+            cy.get('[data-testid="custom-select-save-button"]').scrollIntoView().click();
+            
+            //add property
+            cy.visit(`${Utils.getAppOrigin()}/publisher/apis/${uuid}/properties`);
+            cy.addProperty("property1","value1",true);
```
[the remark this answers is not in the document]
Comment (RusJaI) [reply tests/cypress/integration/publisher/019-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js:89]: solved

**Pull request merged**